agent/agent_endpoint.go

```diff
@@ -1222,7 +1210,25 @@ func (s *HTTPHandlers) AgentRegisterService(resp http.ResponseWriter, req *http.
 		if err := s.agent.AddService(addReq); err != nil {
 			return nil, err
 		}
+
+		ns.Proxy.LocalServicePort = sidecar.Port
+		ns.Proxy.LocalServiceAddress = sidecar.Address
+		ns.Proxy.DestinationServiceID = sidecar.ID
+		ns.Proxy.DestinationServiceName = sidecar.Service
 	}
+
+	addReq := AddServiceRequest{
+		Service:               ns,
+		chkTypes:              chkTypes,
+		persist:               true,
+		token:                 token,
+		Source:                ConfigSourceRemote,
+		replaceExistingChecks: replaceExistingChecks,
+	}
+	if err := s.agent.AddService(addReq); err != nil {
+		return nil, err
+	}
```

Comment:
This block was moved previously, right?

Reply:
This registration block is now done after the sidecar registration, hence moved at the end. Same principle as what has been done in function load_services()